src/mapper.rs

```diff
@@ -0,0 +1,170 @@
+use std::collections::HashMap;
+use std::{borrow::Cow, fmt::Write};
+
+use crate::mapping::{parse_mapping_line, MappingRecord};
+use crate::stacktrace::{parse_stacktrace_line, StackFrame};
+
+#[derive(Debug)]
+struct MemberMapping<'s> {
+    startline: usize,
+    endline: usize,
+    original_class: Option<&'s str>,
+    original: &'s str,
+    original_startline: usize,
+    original_endline: Option<usize>,
+}
+
+#[derive(Debug)]
+struct ClassMapping<'s> {
+    original: &'s str,
+    obfuscated: &'s str,
+    members: HashMap<&'s str, Vec<MemberMapping<'s>>>,
```

Comment:
At least for `members`, you have a rather small number of entries. `BTreeMap` will most likely be more efficient here. For the `classes` list, using a `HashMap` should be fine, though.